packages/destination-actions/src/destinations/salesforce/account/index.ts

```
@@ -170,20 +173,31 @@ const action: ActionDefinition<Settings, Payload> = {
       if (!payload.name) {
         throw new IntegrationError('Missing name value', 'Misconfigured required field', 400)
       }
-      return await sf.createRecord(payload, 'Account')
+      return await sf.createRecord(payload, OBJECT_NAME)
     }
 
     validateLookup(payload)
 
     if (payload.operation === 'update') {
-      return await sf.updateRecord(payload, 'Account')
+      return await sf.updateRecord(payload, OBJECT_NAME)
     }
 
     if (payload.operation === 'upsert') {
       if (!payload.name) {
         throw new IntegrationError('Missing name value', 'Misconfigured required field', 400)
       }
-      return await sf.upsertRecord(payload, 'Account')
+      return await sf.upsertRecord(payload, OBJECT_NAME)
+    }
+  },
+  performBatch: async (request, { settings, payload }) => {
+    const sf: Salesforce = new Salesforce(settings.instanceUrl, request)
+
+    if (payload[0].operation === 'bulkUpsert') {
```

What happens if the operation is not bulkUpsert and we hit performBatch?

---

I think may be a good idea to add an so it doesn't "silently fail" just in case someone misconfigures it.

```js
else {
//  throw IntegrationError ...
}
```